sale_partner_delivery_window/models/res_partner.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from datetime import datetime, timedelta
+
+from odoo import _, models
+from odoo.exceptions import UserError
+from odoo.tools.date_utils import date_range
+
+
+class ResPartner(models.Model):
+
+    _inherit = "res.partner"
+
+    def next_delivery_window_start_datetime(self, from_date=None):
+        """Returns the next starting datetime in a preferred delivery window.
+        If from_date is already in a delivery window, from_date is "the next"
+
+        :param from_date: Datetime object (Leave empty to use now())
+        :return: Datetime object
+        """
+        self.ensure_one()
+        if not from_date:
+            from_date = datetime.now()
+        if self.is_in_delivery_window(from_date):
```

Comment:
Starting from here you should convert from_date that is in UTC to partner TZ

Reply:
I guess this must be fixed with https://github.com/OCA/stock-logistics-workflow/pull/642#discussion_r408194650